src/azure-cli-core/azure/cli/core/commands/arm.py

```diff
@@ -874,9 +876,9 @@ def _update_instance(instance, part, path):  # pylint: disable=too-many-return-s
 
         if hasattr(instance, make_snake_case(part)):
             return getattr(instance, make_snake_case(part), None)
-        elif hasattr(instance, 'additional_properties') and (part in instance.additional_properties):
+        elif part in getattr(instance, 'additional_properties', {}):
             instance.enable_additional_properties_sending()
-            return instance.additional_properties.get[part]
+            return instance.additional_properties[part]
```

Comment:
@tjprescott, this reveals we don't have enough test coverage on this. I will add a new test. At the same time can you take a look at the rest to see whether I am applying the pattern correctly?